dbdimp.h

```diff
@@ -110,22 +124,22 @@
  */
 
 /* Use mysql_options with MYSQL_OPT_SSL_VERIFY_SERVER_CERT */
-#if ((MYSQL_VERSION_ID >= 50023 && MYSQL_VERSION_ID < 50100) || MYSQL_VERSION_ID >= 50111) && (MYSQL_VERSION_ID < 80000 || defined(MARIADB_BASE_VERSION))
+#ifdef CLIENT_SSL_VERIFY_SERVER_CERT
```

Comment:
But this check is for defining `HAVE_SSL_VERIFY`. And it means DBD::mysql can use mysql_options() with `MYSQL_OPT_SSL_VERIFY_SERVER_CERT`. It has nothing with `CLIENT_SSL_VERIFY_SERVER_CERT` flag or mysql_real_connect() function.

Some of those mysql_* SSL related functions are broken in particular MySQL versions and therefore we need to correctly ifdef them and use safe variant.

See https://github.com/perl5-dbi/DBD-mysql/pull/114 and https://github.com/perl5-dbi/DBD-mysql/issues/110

Your change break SSL verification for some MySQL versions.